src/Spatial/Euclidean/StraightLine2D.cs

```diff
@@ -79,61 +60,65 @@ public StraightLine2D(IEnumerable<Point2D> points) : this(points.ToArray())
+        public static StraightLine2D CreateFromCollinear(Point2D[] points)
         {
             if (points.Length < 2)
             {
                 throw new ArgumentException("A line must be constructed from at least two points");
             }
 
-            Gradient = (points[1].Y - points[0].Y) / (points[1].X - points[0].X);
-            if (double.IsInfinity(Gradient))
-            {
-                yoffset = double.NaN; // special case vertical line has no y-intercept
-                xoffset = points[1].X;
-            }
-            else
-            {
-                yoffset = points[1].Y - (Gradient * points[1].X);
-                xoffset = double.NaN;
-            }
-
+            double gradient = (points[1].Y - points[0].Y) / (points[1].X - points[0].X);
+            StraightLine2D line = new StraightLine2D(gradient, points[0].X, points[0].Y);
             for (int i = 2; i < points.Length; i++)
             {
-                if (!IsThrough(points[i]))
+                if (!line.IsThrough(points[i]))
                 {
                     throw new ArgumentException("The points provided are not collinear");
                 }
             }
+
+            return line;
         }
 
         /// <summary>
-        /// Determines if the line is horizontal
+        /// Constructs a straightline
         /// </summary>
-        /// <returns>True if the line is horizontal</returns>
-        public bool IsHorizontal()
+        private StraightLine2D(double gradient, double x, double y)
         {
-            return Gradient == 0;
+            Gradient = gradient;
+            if (double.IsInfinity(gradient))
+            {
+                this.yoffset = double.NaN; // special case vertical line has no y-intercept
+                this.xoffset = x;
+            }
+            else
+            {
+                this.yoffset = y - (gradient * x);
+                this.xoffset = double.NaN;
+            }
         }
 
+        /// <summary>
+        /// Determines if the line is horizontal
+        /// </summary>
+        /// <returns>True if the line is horizontal</returns>
+        public bool IsHorizontal => Gradient == 0;
+
         /// <summary>
         /// Determines if the line is vertical
         /// </summary>
         /// <returns>returns true if the line is vertical</returns>
-        public bool IsVertical()
-        {
-            return double.IsInfinity(Gradient);
-        }
+        public bool IsVertical => double.IsInfinity(Gradient);
 
         /// <summary>
         /// Tests if the other line is parallel to this one
         /// </summary>
         /// <returns>Returns true if lines are parallel</returns>
-        public bool IsParallel(ILine otherLine)
+        public bool IsParallel<T>(T otherLine) where T : ILine2D
         {
             if (otherLine is StraightLine2D)
             {
-                IsParallel((StraightLine2D)otherLine);
+                IsParallel(otherLine);
```

Comment:
Can remove the if now right?

Reply:
if its another type of line say comparing a straightline with a parabola, which is a valid thing to do but should be false, so I think the if has to remain.

Reply:
Ok, I'm not sure `ILine2D` should have this method then but I don't know your plans for it yet.